File: pset1/greedy.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void) {
    float owed;
    int count = 0;
    int round_owed;
    
    printf("O hai! How much change is owed?\n");
    owed = get_float();
    while(owed < 0){
        printf("How much change is owed?\n");
        owed = get_float();
    }
    round_owed = round(owed * 100);
    
    while(round_owed / 25 > 0) {
        round_owed -= 25;
        count++;
    }
    
    while(round_owed / 10 > 0) {
        round_owed -= 10;
        count++;
    }
    
    while(round_owed / 5 > 0) {
        round_owed -= 5;
        count++;
    }
    
    while(round_owed / 1 > 0) {
        round_owed -= 1;
        count++;
    }
    printf("%i\n", count);
}

File: pset6/greedy.py
import cs50

while True:
    print("O hai! How much change is owed?")
    owed = cs50.get_float()
    if owed > 0:
        break;
count = 0;
round_owed = round(owed * 100);
while round_owed // 25 > 0:
    round_owed -= 25
    count += 1

while round_owed // 10 > 0:
    round_owed -= 10
    count += 1

while round_owed // 5 > 0:
    round_owed -= 5
    count += 1

while round_owed // 1 > 0:
    round_owed -= 1
    count += 1

print(count)
